### ISSUE TITLE
CompletionStage with null value should always yield 204

### ISSUE THREAD

**Issue by dandew** [eclipse-ee4j/jersey#3901]
Consider the following:

    @Path("/null")
    @GET
    public String getNull() {
        return null;
    }

    @Path("/null-async")
    @GET
    public CompletionStage<String> getNullAsync() {
        return CompletableFuture.completedFuture(null);
    }

    @Path("/null-async-defer")
    @GET
    public CompletionStage<String> getNullAsyncDefer() {
        final CompletableFuture<String> cf = new CompletableFuture<>();
        executor.schedule(() -> cf.complete(null), 1, SECONDS);
        return cf;
    }

The response statuses for these resources are respectively: `204`, `204`, `200`. The latter should be consistent with the others, i.e. a `204`.

**Comment by spericas** [comment]
Makes perfect sense to me.

**Comment by senivam** [comment]
yes, this sounds reasonable, I was looking at this issue yesterday for now I can only confirm that it acts as described. I will proceed with this to fix the response status.

**Comment by senivam** [comment]
related to #2266

**Comment by terrettaz** [comment]
We are using jersey 2.33 and we encounter the same issue
